packages/integrations/cloudflare/src/server.directory.ts

```diff
@@ -38,17 +29,35 @@ export function createExports(manifest: SSRManifest) {
 				Symbol.for('astro.clientAddress'),
 				request.headers.get('cf-connecting-ip')
 			);
+
+			// @deprecated: getRuntime() can be removed, use `Astro.locals.runtime` instead
 			Reflect.set(request, Symbol.for('runtime'), {
-				...runtimeEnv,
+				...context,
 				waitUntil: (promise: Promise<any>) => {
-					runtimeEnv.waitUntil(promise);
+					context.waitUntil(promise);
 				},
 				name: 'cloudflare',
 				next,
 				caches,
 				cf: request.cf,
 			});
-			let response = await app.render(request, routeData);
+
+			let response = await app.render(request, routeData, {
+				env: env,
+				cf: request.cf,
+				runtime: {
+					// request: Request; // not needed because we have Astro.request, thus they are minor differences
+					// functionPath: string; // not needed
+					waitUntil: (promise: Promise<any>) => { context.waitUntil(promise); },
+					// passThroughOnException: () => void; // probably not needed ??
+					// next: (input?: Request | string, init?: RequestInit) => Promise<Response>; // probably not needed ??
+					env: context.env,
+					params: context.params, // Isn't that the same as Astro.props ??
```

Comment:
Are Astro.props 1-1 equally? Haven't found anything.

Reply:
No. `params` come from the URL the page. `props` belongs to a component.

Reply:
My bad, I meant [Astro.params](https://docs.astro.build/en/reference/api-reference/#astroparams)